### PR TITLE
Make map name less unforgiving

### PR DESCRIPTION
**Pull request by simon33-2**
Similar to #1651 but keeps the mapName property alive - just removes arbitrary restrictions on what it can be called - as well as trying the mangled names, also try the unmangled names. Should get rid of that extremely frustrating message after downloading and starting a map - "Please download the map".

### CODE REVIEW

File: src/main/java/games/strategy/triplea/ResourceLoader.java [hunk 1]
```diff
@@ -97,6 +97,8 @@ protected static String normalizeMapZipName(final String zipName) {
 
     // clicking github 'clone or download' and downloading the zip gives a zip that ends with "-master.zip"
     candidates.add(new File(ClientFileSystemHelper.getUserMapsFolder(), normalizeMapZipName(mapName) + "-master.zip"));
+    candidates.add(new File(ClientFileSystemHelper.getUserMapsFolder(), mapName + "-master.zip"));
+    candidates.add(new File(ClientFileSystemHelper.getUserMapsFolder(), zipName));
```
Comment (simon33-2) [inline src/main/java/games/strategy/triplea/ResourceLoader.java]: How did I not notice that? I'll update.

**Comment by simon33-2** [comment]
There already are tests. Perhaps I can add one. At least it makes it harder to break it again later.

File: src/main/java/games/strategy/engine/data/GameParser.java [hunk 2]
```diff
@@ -589,21 +605,21 @@ private void setGrids(final GameData data, final String gridType, final String n
...
       diagonalConnectionsImplict = false;
     } else {
-      throw new GameParseException(mapName,
+      throw new GameParseException(mapPathName,
           "diagonal-connections attribute must be either \"explicit\" or \"implicit\"");
     }
-    final int x_size = Integer.valueOf(xs);
-    int y_size;
+    final int xSize = Integer.valueOf(xs);
+    int yySize;                                 // ySize or y_size == checkstyle violation
```
Comment (DanVanAtta) [inline src/main/java/games/strategy/engine/data/GameParser.java]: `yySize` and `xSize` don't work well as name pairs.
When viewing this code without a diff ` // ySize or y_size == checkstyle violation` will not make very much sense. Why not yet more variations, like `my_ySize`, or int_ySize?

Looking down a few lines, I see:
`map.setGridDimensions(xSize, yySize);`

based on that, some name recommendations:
xSize => mapGridWidth
yySize => mapGridHeight

or
xSize => mapGridColCount
yySize => mapGridRowCount

File: src/main/java/games/strategy/engine/data/GameParser.java [hunk 3]
```diff
@@ -926,14 +942,15 @@ private void parseProperties(final Node root) throws GameParseException {
       data.getProperties()
           .addEditableProperty(new NumberProperty(Constants.AI_BONUS_DEFENSE, null, data.getDiceSides(), 0, 0));
     }
+    properties.set(Constants.MAP_NAME, mapCodeName);    // Overwrite map name with containing folder
```
Comment (DanVanAtta) [inline src/main/java/games/strategy/engine/data/GameParser.java]: I'm a bit confused by the naming:

`mapCodeName` +  `// Overwrite map name with containing folder`

Does that mean that `mapCodeName` == `mapFolder`? or `mapContainingFolder`?

Comment (DanVanAtta) [inline src/main/java/games/strategy/engine/data/GameParser.java]: Could you also provide some pointers on how we are consuming this property @simon33-2 ? I'm interested if we can make that relationship more direct.

**Comment by simon33-2** [comment]
Re: mapCodeName = MapPathParser.extractMapCodeName(pathName);

Where is this MapPathParser? git grep and find don't reveal it?

**Comment by simon33-2** [comment]
I agree that all the checkstyle stuff makes this PR very confusing. I think I couldn't get past the check without fixing some of the checkstyle changes.

Do you have any suggestions for what the test you are asking for should actually do?

**Comment by simon33-2** [comment]
Took me a while to remember the why. The purpose was to find out the directory or zip name and not necessarily rely on the map name parameter.

Perhaps all that is needed is the one line change to ResourceLoader:
 +    candidates.add(new File(ClientFileSystemHelper.getUserMapsFolder(), mapName + "-master.zip"));

**Comment by ssoloff** [comment]
> Perhaps all that is needed is the one line change to ResourceLoader:
>
> * candidates.add(new File(ClientFileSystemHelper.getUserMapsFolder(), mapName + "-master.zip"));

Can this candidate ever occur?

My understanding is that the local name of maps downloaded from GitHub zipballs is calculated in `DownloadFileDescription#getInstallLocation()`:

```java
  public File getInstallLocation() {
    String masterSuffix = (getMapZipFileName().toLowerCase().endsWith("master.zip")) ? "-master" : "";
    String normalizedMapName = getMapName().toLowerCase().replace(' ', '_') + masterSuffix + ".zip";
    return new File(ClientFileSystemHelper.getUserMapsFolder() + File.separator + normalizedMapName);
  }
```

So that the "map name" portion of the zip file name will always be normalized (albeit not using the same algorithm as `ResourceLoader`).  Thus the existing candidate:

```java
candidates.add(new File(ClientFileSystemHelper.getUserMapsFolder(), normalizeMapZipName(mapName) + "-master.zip"));
```

should be sufficient, right?  Is there a different code path that can result in using a non-normalized map name for GitHub zipballs?

Downloading the zipball directly from GitHub also results in a normalized map name (well, it's the name of the repo, which seems to follow this particular naming convention throughout all the `triplea-maps` repos).

**Comment by simon33-2** [comment]
> Can this candidate occur?

Well we still have had issues with mismatches. You're correct though - just tested it and it seems that it doesn't really work this way.

**Pull request closed without merging**